code/game/objects/structures/statues.dm

```diff
@@ -293,10 +253,212 @@
+		preset_statue.set_custom_materials(prepared_block.custom_materials)
+		QDEL_NULL(prepared_block)
+	else if(!QDELETED(current_target))
+		var/obj/structure/statue/custom/new_statue = new(get_turf(prepared_block))
+		new_statue.set_visuals(current_target)
+		new_statue.set_custom_materials(prepared_block.custom_materials)
+		new_statue.name = "statue of [current_target]"
+		new_statue.desc = "statue depicting [current_target]"
+		QDEL_NULL(prepared_block)
+	current_target = null
+	current_preset_type = null
+
+
+/obj/item/chisel/proc/show_generic_statues_prompt(mob/living/user)
+	var/list/choices = list()
+	for(var/statue_path in prepared_block.get_possible_statues())
+		var/obj/structure/statue/S = statue_path
+		choices[statue_path] = image(icon=initial(S.icon),icon_state=initial(S.icon_state))
+	var/choice = show_radial_menu(user, prepared_block , choices, require_near = TRUE)
+	if(choice)
+		current_preset_type = choice
+		var/obj/structure/statue/S = choice
+		to_chat(user,"<span class='notice'>You decide to sculpt [prepared_block] into [initial(S.name)].</span>")
+		current_target = null
+
+/obj/item/chisel/proc/is_viable_target(atom/movable/AM)
+	//Only things on turfs
+	if(!isturf(AM.loc))
+		return FALSE
+	//No big icon things
+	var/icon/thing_icon = icon(AM.icon, AM.icon_state)
+	if(thing_icon.Height() != world.icon_size || thing_icon.Width() != world.icon_size)
+		return FALSE
+	return TRUE
+
+/obj/structure/carving_block
+	name = "block"
+	desc = "ready for sculpting."
+	icon = 'icons/obj/statue.dmi'
+	icon_state = "block"
+	material_flags = MATERIAL_COLOR | MATERIAL_ADD_PREFIX
+	density = TRUE
+
+	//Table of required materials for each non-abstract statue type
+	var/static/list/statue_costs
+
+/// Returns a list of preset statues carvable from this block depending on the custom materials
+/obj/structure/carving_block/proc/get_possible_statues()
+	. = list()
+	if(!statue_costs)
+		statue_costs = build_statue_cost_table()
+	statue_loop:
+		for(var/statue_path in statue_costs)
+			var/list/carving_cost = statue_costs[statue_path]
+			for(var/required_material in carving_cost)
+				if(!custom_materials[required_material] || custom_materials[required_material] < carving_cost[required_material])
+					continue statue_loop
+			. += statue_path
+
+/obj/structure/carving_block/proc/build_statue_cost_table()
+	. = list()
+	for(var/statue_type in subtypesof(/obj/structure/statue) - /obj/structure/statue/custom)
+		var/obj/structure/statue/S = new statue_type()
+		if(!S.icon_state || S.abstract_type == S.type || !S.custom_materials)
+			continue
+		.[S.type] = S.custom_materials
+		qdel(S)
+
+
+/obj/structure/statue/custom
+	name = "custom statue"
+	icon_state = "base"
+	obj_flags = CAN_BE_HIT | UNIQUE_RENAME
+	appearance_flags = TILE_BOUND | PIXEL_SCALE | KEEP_TOGETHER //Added keep together in case targets has weird layering
+	material_flags = MATERIAL_COLOR
```

Comment:
```suggestion
	material_flags = MATERIAL_COLOR | MATERIAL_ADD_PREFIX | MATERIAL_AFFECT_STATISTICS
```
1. This makes the statue "gold x statue"
2. Makes wooden statues etc actually become flammable.